src/net/send_recv/msg.rs

```diff
@@ -583,11 +584,11 @@ mod messages {
 
     impl<'buf> Messages<'buf> {
         /// Create a new iterator over messages from a byte buffer.
-        pub(super) fn new(buf: &'buf mut [u8], len: usize) -> Self {
+        pub(super) fn new(buf: &'buf mut [u8]) -> Self {
             let msghdr = {
                 let mut h: c::msghdr = unsafe { core::mem::zeroed() };
                 h.msg_control = buf.as_mut_ptr().cast();
-                h.msg_controllen = len as _;
+                h.msg_controllen = crate::backend::msg_control_len(buf.len());
```

Comment:
I was expecting this would want to fail on overflow, but `msg_control_len` does a saturation. Is that intended?

Reply:
Good catch, thanks!